controllers/etcd_controller.go

```diff
@@ -414,59 +357,90 @@ func (r *EtcdReconciler) delete(ctx context.Context, etcd *druidv1alpha1.Etcd) (
 
 func (r *EtcdReconciler) reconcilePodDisruptionBudget(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, values map[string]interface{}) error {
 	logger.Info("Reconcile PodDisruptionBudget")
-	pdb := &policyv1beta1.PodDisruptionBudget{}
-	err := r.Get(ctx, types.NamespacedName{Name: etcd.Name, Namespace: etcd.Namespace}, pdb)
-
-	if err == nil {
-		// pdb already exists, claim it
-
-		selector, err := metav1.LabelSelectorAsSelector(etcd.Spec.Selector)
-		if err != nil {
-			logger.Error(err, "Error converting etcd selector to selector")
-			return err
-		}
 
-		logger.Info("Claiming pdb object")
-		_, err = r.claimPodDisruptionBudget(ctx, etcd, selector, pdb)
+	// Find out k8s version
+	targetCoreClient, err := k8s.NewForConfig(r.Config)
+	if err != nil {
 		return err
 	}
-
-	if !apierrors.IsNotFound(err) {
+	targetServerVersion, err := targetCoreClient.Discovery().ServerVersion()
+	if err != nil {
 		return err
 	}
-
-	// Required podDisruptionBudget doesn't exist. Create new
-	pdb, err = r.getPodDisruptionBudgetFromEtcd(etcd, values, logger)
+	targetKubernetesVersion, err := semver.NewVersion(targetServerVersion.GitVersion)
 	if err != nil {
 		return err
 	}
 
-	logger.Info("Creating PodDisruptionBudget", "poddisruptionbudget", kutil.Key(pdb.Namespace, pdb.Name).String())
-	if err := r.Create(ctx, pdb); err != nil {
-		return err
-	}
+	if k8sutils.ConstraintK8sGreaterEqual121.Check(targetKubernetesVersion) {
+		pdb := &policyv1.PodDisruptionBudget{}
 
-	return nil
-}
+		err := r.Get(ctx, types.NamespacedName{Name: etcd.Name, Namespace: etcd.Namespace}, pdb)
+		if err == nil {
+			// pdb already exists
+			return nil
+		}
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
 
-func (r *EtcdReconciler) getPodDisruptionBudgetFromEtcd(etcd *druidv1alpha1.Etcd, values map[string]interface{}, logger logr.Logger) (*policyv1beta1.PodDisruptionBudget, error) {
-	var err error
-	decoded := &policyv1beta1.PodDisruptionBudget{}
-	pdbPath := getChartPathForPodDisruptionBudget()
-	chartPath := getChartPath()
-	renderedChart, err := r.chartApplier.Render(chartPath, etcd.Name, etcd.Namespace, values)
-	if err != nil {
-		return nil, err
-	}
-	if content, ok := renderedChart.Files()[pdbPath]; ok {
-		decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(content)), 1024)
-		if err = decoder.Decode(&decoded); err != nil {
-			return nil, err
+		// Required podDisruptionBudget doesn't exist. Create new
+		pdbPath := getChartPathForPodDisruptionBudget()
```

Comment:
the chart contains `v1beta1` for PDB, will this create a PDB version version = v1 when k8s version > 1.21 and there is no existing PDB? Can you double check this?

Reply:
PDBs are now created using components, so this should be resolved now